### DISCUSSION TITLE
Error: No such command 'run'

### DISCUSSION THREAD

**Discussion by IamSandeepGunda** [kedro-org/kedro#946] [Idea]
As a part of upgrading Kedro from 0.16.2 to 0.17.3 in our organization, I've made changes to Kedro related files in our codebase based on Kedro starter pyspark-iris on 0.17.3. 

Now I get an error of **Error: No such command 'run'** on kedro run. 

I've posted my question on Stack overflow with code sample because it had bvetter code formatting. 
[Link to Stack Overflow question](https://stackoverflow.com/questions/69499388/kedro-cannot-find-run)

**Comment by datajoely** [comment]
Happy to help you work through this. Have you had a chance to follow the upgrading from 0.16.x to 0.17.x ?

https://github.com/quantumblacklabs/kedro/blob/master/RELEASE.md#migration-guide-from-kedro-016-to-017

Also would you mind opening issues in just one channel in the future? Officially we only support GitHub discussions and Discord - StackOverflow posts appear in Discord via RSS for the community to pick up.

**Reply by datajoely** [reply]
Additionally any reason you're tied to 0.17.3 not the latest 0.17.5 release?

**Reply by IamSandeepGunda** [reply]
we started upgrading kedro across various projects one by one. So the the project which upgraded it first went to 0.17.3 so we followed the suite. It maybe due to limitations I'm unaware, or just 0.17.3 being available then.

**Reply by datajoely** [reply]
Understood - I was thinking of a nasty bug where we were swallowing an exception, but it was fixed in 0.17.3 so that's not a problem here. 

It's late in the day here in London, so let us know if that migration helps and we'll pick up any updates on Monday. Have a good weekend!

**Reply by IamSandeepGunda** [reply]
Sure, let me see if I missed anything in that link. But I've already migrated one project to 0.17.3 before helping with this. And everything seems pretty identical to that one so no idea why one works while the other doesn't. Let's pick up from where we are on Monday.

**Comment by datajoely** [comment]
Hi @IamSandeepGunda I was thinking about this - it may be come down to the presence of the `pyproject.toml` file which was introduced in 0.17.0:

If you run `kedro run` outside of a project you get this error:
![image](https://user-images.githubusercontent.com/35801847/136767298-5af125dc-8692-417a-bf12-821c94d258c4.png)

However, inside a project directory we use the contents of `pyproject.toml` as the 'entry point' into a kedro project:

![image](https://user-images.githubusercontent.com/35801847/136767357-0c1e28ba-0ae4-4ca0-9a73-fc671bd368af.png)

Within this folder the run command is now available:
![image](https://user-images.githubusercontent.com/35801847/136767489-87ffb97c-9415-4692-b110-9790cf33875f.png)

**Reply by IamSandeepGunda** [reply]
I'd say that's pretty accurate! I got to the root cause late Friday. Our dockerfile was installing a dependent kedro project along with it's requirements after our project's requirements were installed. And the dependent project had kedro==0.16.4 in it's requirements so the kedro version on the container was getting downgraded. Hence Kedro -0.16.4 never knew run got relocated. Thanks for your help though @datajoely!

**Comment by regisnf** [comment]
I'm getting the same error from command line when trying to configure vscode (kedro v0.17.6) as in documentation: https://kedro.readthedocs.io/en/stable/09_development/01_set_up_vscode.html: "no such command 'run'. 

Any ideas on how to setup VSCode now?

**Reply by datajoely** [reply]
This means the python environment that VS Code is using doesn't point to one with Kedro installed.
